### ISSUE TITLE
Space complexity of solution to Maximum Depth Of Binary Tree

### ISSUE THREAD

**Issue by snapfinger** [soapyigu/LeetCode-Swift#301]
The space complexity was specified as O(1) in the [solution](https://github.com/soapyigu/LeetCode-Swift/blob/master/Tree/MaximumDepthOfBinaryTree.swift), however because of recursion, considering the stack, should it be O(n) instead?

**Comment by KartikSatoskar** [comment]
@snapfinger Yes you are correct. Space complexity of the solution should be O(n). 
@soapyigu Should I raise the PR to fix this?
